src/components/map/layers/edit/EditLinkLayer.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react';
+import * as L from 'leaflet';
+import { withLeaflet } from 'react-leaflet';
+import { matchPath } from 'react-router';
+import { inject, observer } from 'mobx-react';
+import { IReactionDisposer, reaction } from 'mobx';
+import navigator from '~/routing/navigator';
+import SubSites from '~/routing/subSites';
+import { INode, ILink } from '~/models';
+import { EditNetworkStore } from '~/stores/editNetworkStore';
+import NodeMarker from '../mapIcons/NodeMarker';
+import { LeafletContext } from '../../Map';
+
+interface IEditLinkLayerProps {
+    editNetworkStore?: EditNetworkStore;
+    leaflet: LeafletContext;
+}
+
+@inject('editNetworkStore')
+@observer
+class EditLinkLayer extends Component<IEditLinkLayerProps> {
+    private reactionDisposers: IReactionDisposer[] = [];
+    private editableLinks: L.Polyline[] = [];
+
+    componentDidMount() {
+        this.reactionDisposers.push(reaction(
```

Comment:
No need for reactionDisposers to have multiple reactions.

Reply:
Using one reactionDisposer now